File: CS/FilteringWithComboBoxEditor/Converter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Windows.Data;

namespace FilteringWithComboBoxEditor
{
    public class TextConverter : IValueConverter
    {
        public object Convert(object value, Type targetType, object parameter, System.Globalization.CultureInfo culture)
        {
            if (value != null && value is ElementItem)
                return (value as ElementItem).ItemName;
            else return "";
        }

        public object ConvertBack(object value, Type targetType, object parameter, System.Globalization.CultureInfo culture)
        {
            if (value != null && value is ElementItem)
                return (value as ElementItem).ItemName;
            else return "";
        }
    }
}

File: VB/FilteringWithComboBoxEditor/Converter.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Windows.Data

Namespace FilteringWithComboBoxEditor
    Public Class TextConverter
        Implements IValueConverter

        Public Function Convert(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As System.Globalization.CultureInfo) As Object Implements IValueConverter.Convert
            If value IsNot Nothing AndAlso TypeOf value Is ElementItem Then
                Return (TryCast(value, ElementItem)).ItemName
            Else
                Return ""
            End If
        End Function

        Public Function ConvertBack(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As System.Globalization.CultureInfo) As Object Implements IValueConverter.ConvertBack
            If value IsNot Nothing AndAlso TypeOf value Is ElementItem Then
                Return (TryCast(value, ElementItem)).ItemName
            Else
                Return ""
            End If
        End Function
    End Class
End Namespace
